src/User.js

```diff
@@ -80,26 +101,76 @@ class User {
       return sum;
     }, 0) / 7).toFixed(1);
   }
+
+  getSleepQualityByDate(date) {
+      // if(!date){
+    //   this.sleepQualityAverage[0].date;
+    // }
+    let valueNeeded = 0;
+    this.sleepQualityRecord.forEach(item => {
+      if (item.date === date)  {
+        valueNeeded = item.quality;
+      }
+    })
+    return valueNeeded;
+  }
+
+  getHoursSleptByDate(date) {
+      // if(!date){
+    //   this.sleepQualityAverage[0].date;
+    // }
+    let valueNeeded = 0;
+    console.log("hours", this.sleepHoursRecord)
+    this.sleepHoursRecord.forEach(item => {
+      if (item.date === date)  {
+        valueNeeded = item.hours;
+      }
+    })
+    return valueNeeded;
+  }
+
+
+
+
+
+
+  ////////////////////////  - ACTIVITY MEHTODS - ////////////////////////
+  // update activity data to corresponding user instances ------>
   updateActivities(activity) {
     this.activityRecord.unshift(activity);
+    // console.log("activity record in user class:", this.activityRecord)
     if (activity.numSteps >= this.dailyStepGoal) {
       this.accomplishedDays.unshift(activity.date);
     }
   }
-  findClimbingRecord() {
-    return this.activityRecord.sort((a, b) => {
-      return b.flightsOfStairs - a.flightsOfStairs;
-    })[0].flightsOfStairs;
+
+  //This method is recyclable and help us for all the activity info for today
+  findActivityInfoToday(user, todayDate) {
```

Comment:
This logic for date was added to the activity methods that required the date as well.